src/main/java/org/elasticsearch/search/aggregations/reducers/movavg/models/MovAvgModel.java

```diff
@@ -90,6 +107,105 @@
      * @throws IOException
      */
     public abstract void writeTo(StreamOutput out) throws IOException;
+
+    /**
+     * Abstract class which provides some concrete parsing functionality.  If the
+     * model doesn't need to parse any parameters, it can implement MovAvgModelParser
+     * directly instead of extending AbstractModelParser
+     */
+    public abstract static class AbstractModelParser implements MovAvgModelParser {
```

Comment:
MovAvgModelParser is an interface.  Do you think I should just ditch the interface and make everything extend an abstract, base class?